src/answers/services/answers.service.ts

```diff
@@ -228,4 +232,48 @@ export class AnswersService extends BaseService<
       }
     });
   }
+
+  async getUrls(answer: AnswerDocument): Promise<AnswerDocument> {
+    const responses = answer.responses;
+    for await (const response of responses) {
+      if (this.responseType(response) === 'url_array') {
+        if (Array.isArray(response.value)) {
+          const newArray: IAnswerFile[] = [];
+          for await (const file of response.value) {
+            const castFile = file as IAnswerFile;
+            newArray.push({
+              url: await this.s3Service.generatePresignedUrl({
+                key: castFile.url,
+              }),
+              isPublic: castFile.isPublic,
+            });
+            response.value = newArray;
+          }
+        }
+      }
+      if (this.responseType(response) === 'url_object') {
+        const castFile = response.value as IAnswerFile;
+        response.value = {
+          url: await this.s3Service.generatePresignedUrl({
+            key: castFile.url,
+          }),
+          isPublic: castFile.isPublic,
+        };
+      }
+    }
+    answer.responses = responses;
+    return answer;
+  }
+
+  responseType(response: IAnswerResponse): string {
+    if (Array.isArray(response.value)) {
+      if (typeof response.value[0] !== 'string' && response.value[0].url)
+        return 'url_array';
+      return 'string_array';
+    }
+
+    if (response.value && typeof response.value !== 'string')
+      return 'url_object';
+    return 'string';
+  }
```

Comment:
Better to have individual, isUrlArrayType, isStringArrayType, isUrlObjectType, and isStringType methods. Also, you may be able to use type predicates to add type information to the inpuit parameter https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates